docs/dataset-sql-explanation.md

```diff
@@ -0,0 +1,50 @@
+## What happens when Data Builder generates a dataset?
```

Comment:
suggestion: Can we link this into the temporary `data-builder-toc.md`, please? So that it's part of the index and we know to add this to `mkdocs.yml` eventually.